src/ui/pages/Settings/Settings.tsx

```diff
@@ -340,6 +343,44 @@ function ToggleSettingLine({
   );
 }
 
+function ClearPendingTransactionsLine() {
+  const toastRef = useRef<PopoverToastHandle>(null);
+  const { mutate: clearPendingTransactions, ...mutation } = useMutation({
+    mutationFn: async () => {
+      await new Promise((r) => setTimeout(r, 500)); // artificial delay
+      toastRef.current?.removeToast();
+      return walletPort.request('clearPendingTransactions');
+    },
+    onSuccess: () => {
+      toastRef.current?.showToast();
+    },
+  });
+  return (
+    <>
+      <PopoverToast ref={toastRef}>Pending transactions cleared</PopoverToast>
+
+      <Frame
+        as={UnstyledButton}
+        interactiveStyles={true}
+        onClick={() => clearPendingTransactions()}
+        disabled={mutation.isLoading}
+        style={{ padding: 20 }}
+      >
+        <Media
```

Comment:
Do we need Media here?

Reply:
Yeah not really. I reused the layout of the `ToggleSettingLine` but I guess I'll refactor it, too, then